bitswap/client/internal/session/session.go

```diff
@@ -304,32 +305,44 @@ func (s *Session) run(ctx context.Context) {
 	for {
 		select {
 		case oper := <-s.incoming:
+			var opCtx context.Context
+			var span trace.Span
 			switch oper.op {
 			case opReceive:
 				// Received blocks
+				trace.SpanFromContext(ctx).AddEvent("Session.ReceiveOp")
 				s.handleReceive(oper.keys)
```

Comment:
It'd be nice to add the keys to the event here, although a little more logic is might be needed because stringifying every CID even when tracing is disabled (or on a minimal fraction of requests) would be a pain.

Reply:
Going to skip this for now